### PR TITLE
Add Provider Account (Tenant) resources '/provider'

### PR DESCRIPTION
**Pull request by jsmolar**
Client and Resource for Provider Account was added. '/provider' endpoint on admin portal.
The name: 'Provider Account' was already taken with its user (Provider Account User). This error is fixed in this PR

### CODE REVIEW

File: threescale_api/resources.py [hunk 1]
```diff
@@ -597,12 +597,25 @@ def url(self) -> str:
         return self.threescale_client.admin_api_url + '/registry/policies'
 
 
-class ProviderAccounts(DefaultStateClient):
+class ProviderAccounts(DefaultClient):
```
Comment (pehala) [inline threescale_api/resources.py:600]: If I understand this correctly, this is a rename (ProviderAccounts -> ProviderAccountUsers), does this affect the test suite in any way?

File: threescale_api/resources.py [hunk 2]
```diff
@@ -1147,7 +1160,12 @@ def service(self) -> 'Service':
         return self.proxy.service
 
 
-class ProviderAccount(DefaultStateResource):
+class ProviderAccount(DefaultResource):
```
Comment (pehala) [inline threescale_api/resources.py:1163]: While not necessary, would it be possible to add tests for this resource?

File: threescale_api/resources.py (hunk 1 above)
Comment (jsmolar) [reply threescale_api/resources.py:600]: Currently, it is used only by the permissions UI test, which is not merged.

File: threescale_api/resources.py (hunk 2 above)
Comment (jsmolar) [reply threescale_api/resources.py:1163]: The only test for this resource would be "list" test, as 3scale endpoint: '/provider' is enabled only for list and update. Uptade is not used and probably will not be used in the future.

**Review by pehala — APPROVED** [review]

**Comment by mkudlej** [comment]
LGTM

**Review by mkudlej — APPROVED** [review]

**Pull request merged**